src/main/java/seedu/address/ui/EventListPanel.java

```diff
@@ -29,26 +31,60 @@
     @FXML
     private ListView<EventCard> eventListView;
 
+    private Label emptyLabel = new Label("Event List is empty!");
+
     public EventListPanel(ObservableList<Activity> eventList) {
         super(FXML);
         setConnections(eventList);
         registerAsAnEventHandler(this);
+        setUpPlaceHolder();
+        //maybe do not need this
+        //eventListView.managedProperty().bind(eventListView.visibleProperty());
     }
 
     private void setConnections(ObservableList<Activity> eventList) {
         ObservableList<EventCard> mappedList = EasyBind.map(
                 eventList, (event) -> new EventCard(event, eventList.indexOf(event) + 1));
         eventListView.setItems(mappedList);
-        eventListView.setCellFactory(listView -> new EventListViewCell());
+        linkCell();
         setEventHandlerForSelectionChangeEvent();
     }
 
+    private void setUpPlaceHolder()   {
+        eventListView.setPlaceholder(emptyLabel);
+        emptyLabel.setStyle("-fx-font-family: \"Open Sans\"; -fx-font-size: 25px; ");
+    }
+
+    /**
+     * Links eventListView to eventListViewCell as its custom ListCell
+     */
+    private void linkCell() {
+        eventListView.setCellFactory(listView -> {
+            EventListViewCell cell = new EventListViewCell();
+            cell.addEventFilter(MouseEvent.MOUSE_PRESSED, event -> {
+                eventListView.requestFocus();
+                if (!cell.isEmpty()) {
+                    int index = cell.getIndex();
+                    if (eventListView.getSelectionModel().getSelectedIndices().contains(index))  {
+                        logger.fine("Selection in event list panel with index '" + index
```

Comment:
We can improve the code quality here by flattening the `arrowhead code` perhaps the link here can help with [this](https://blog.codinghorror.com/flattening-arrow-code/)? 
We can additional methods to flatten the arrowhead code too =).